Sources/Apollo/HTTPNetworkTransport.swift

```diff
@@ -105,106 +112,166 @@ public class HTTPNetworkTransport: NetworkTransport {
+          let retryByDefault: Bool
+          if let reason = reason as? GraphQLHTTPResponseError,
+            reason.kind == apqsError.kind {
+            // if the same error occurs, respect retry option from delegate
+            retryByDefault = false
+          } else {
+            retryByDefault = true
+          }
+          // Auto Persisted Query handling
+          self.handleErrorOrRetry(operation: operation,
+                                  error: apqsError,
+                                  for: request,
+                                  response: response,
+                                  retryByDefault: retryByDefault,
+                                  completionHandler: completionHandler)
+
+        }else {
+          // no errors
+          let response = GraphQLResponse(operation: operation, body: body)
+          completionHandler(response, nil)
+        }
       } catch let parsingError {
-        self?.handleErrorOrRetry(operation: operation,
-                                 error: parsingError,
-                                 for: request,
-                                 response: response,
-                                 completionHandler: completionHandler)
+        self.handleErrorOrRetry(operation: operation,
+                                error: parsingError,
+                                for: request,
+                                response: response,
+                                completionHandler: completionHandler)
+        
       }
     }
-    
     task.resume()
-    
     return task
   }
-
-  private let sendOperationIdentifiers: Bool
   
+  // retryByDefault: still retry if delegate is absent, otherwise repect the value from delegate
   private func handleErrorOrRetry<Operation>(operation: Operation,
                                              error: Error,
                                              for request: URLRequest,
                                              response: URLResponse?,
+                                             retryByDefault: Bool = false,
                                              completionHandler: @escaping (_ response: GraphQLResponse<Operation>?, _ error: Error?) -> Void) {
-    guard
-      let delegate = self.delegate,
-      let retrier = delegate as? HTTPNetworkTransportRetryDelegate else {
+    if let delegate = self.delegate,
+      let retrier = delegate as? HTTPNetworkTransportRetryDelegate {
+      retrier.networkTransport(
+        self,
+        receivedError: error,
+        for: request,
+        response: response,
+        retryHandler: { [weak self] shouldRetry in
+          guard let self = self, shouldRetry else {
+            completionHandler(nil, error)
+            return
+          }
+          _ = self.send(operation: operation, retryFor: error, completionHandler: completionHandler)
+      })
+    } else if retryByDefault {
+      _ = self.send(operation: operation, retryFor: error, completionHandler: completionHandler)
```

Comment:
I think it would be clearer to check if APQ is enabled && the error is an APQ error first, then either re-send the query if it is, or continue going through and checking for the delegate if it's not. It'd probably mean moving the `retry` logic from that second `send` method down here - which think is probably clearer in the end.